scripts/monitor.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+source .env
+
+log() {
+    echo "$(date +'%Y-%m-%d %H:%M:%S') - $1"
+}
+
+send_alert() {
+    message=$1
+    logs=$2
+    echo "Sending alert to slack $CHANNEL ... $message"
+
+    curl -X POST "https://slack.com/api/chat.postMessage" \
+        -H "Authorization: Bearer $BOT_AUTH_TOKEN" \
+        -H 'Content-type: application/json; charset=utf-8' \
+        --data '{
+            "channel": "'"$CHANNEL"'",
+            "text": "'"$message"'\n'"$logs"'"
+        }' > /dev/null 2>&1
+}
+
+get_container_names() {
+    docker ps --format "{{.ID}}\t{{.Names}}" | grep -v "$(hostname)" | awk '{print $2}'
+}
+
+check_container_status() {
+    containers=$(get_container_names)
+    if [ "$containers" == "" ]; then
+        echo "No containers available"
+        return 1
+    fi
+
+    for container in $containers; do
+        status=$(
+            docker inspect \
+                --format="{{if .State.Health}}{{.State.Health.Status}}{{else}}{{.State.Status}}{{end}}" \
+                "$container"
+        )
+        if [ "$status" != "healthy" ] && [ "$status" != "running" ]; then
+            output=$(docker inspect "$container")
+            return 1
+        fi
+    done
+    return 0
+}
+
+check_logs_for_error() {
+    # Tail the latest `limit` number of lines in the docker log. 100 lines is roughly 1 minute worth
+    # of logs, so we scale `limit` based on the interval (2x) to ensure we don't miss any logs.
+    # Additionally, we cut out logs older that `CHECK_INTERVAL_SECS` seconds so that we don't match
+    # on older errors.
+    limit=$(( 2 * CHECK_INTERVAL_SECS ))
+
+    for container in $(get_container_names); do
+        if output=$(
+            docker logs --tail $limit "$container" | \
+                grep -A999999 "$(date -u -d @$(($(date -u +%s) - CHECK_INTERVAL_SECS)) +'%Y-%m-%dT%H:%M:')" | \
```

Comment:
For this part I guess better we need some discussion.
Normally we prefer we can get the last 50 lines of a crashed service log.
Not by keywords like "ERROR" as in fact some times like the mongodb failed with no any error keywords.

If there is some limitation that cannot send too much charactors 50 line to slack channel we may need think about other method like send attached txt files.